src/main/java/org/apache/cassandra/sidecar/routes/restore/RestoreJobProgressHandler.java

```diff
@@ -0,0 +1,88 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.sidecar.routes.restore;
+
+import java.util.List;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.core.net.SocketAddress;
+import io.vertx.ext.web.RoutingContext;
+import org.apache.cassandra.sidecar.common.ApiEndpointsV1;
+import org.apache.cassandra.sidecar.common.data.RestoreJobProgressFetchPolicy;
+import org.apache.cassandra.sidecar.common.utils.Preconditions;
+import org.apache.cassandra.sidecar.concurrent.ExecutorPools;
+import org.apache.cassandra.sidecar.db.RestoreJob;
+import org.apache.cassandra.sidecar.restore.RestoreJobConsistencyLevelChecker;
+import org.apache.cassandra.sidecar.restore.RestoreJobProgress;
+import org.apache.cassandra.sidecar.routes.AbstractHandler;
+import org.apache.cassandra.sidecar.routes.RoutingContextUtils;
+import org.apache.cassandra.sidecar.utils.CassandraInputValidator;
+import org.apache.cassandra.sidecar.utils.InstanceMetadataFetcher;
+
+import static org.apache.cassandra.sidecar.routes.RoutingContextUtils.SC_RESTORE_JOB;
+
+/**
+ * Provides a REST API for querying the progress of a {@link RestoreJob}
+ * The response content can vary based on the {@link RestoreJobProgressFetchPolicy}
+ */
+@Singleton
+public class RestoreJobProgressHandler extends AbstractHandler<RestoreJobProgressFetchPolicy>
+{
+    private final RestoreJobConsistencyLevelChecker consistencyLevelChecker;
+
+    /**
+     * Constructs a handler with the provided {@code metadataFetcher}
+     *
+     * @param metadataFetcher the interface to retrieve instance metadata
+     * @param executorPools   the executor pools for blocking executions
+     * @param validator       a validator instance to validate Cassandra-specific input
+     */
+    @Inject
+    public RestoreJobProgressHandler(InstanceMetadataFetcher metadataFetcher,
+                                     ExecutorPools executorPools,
+                                     CassandraInputValidator validator,
+                                     RestoreJobConsistencyLevelChecker consistencyLevelChecker)
+    {
+        super(metadataFetcher, executorPools, validator);
+        this.consistencyLevelChecker = consistencyLevelChecker;
+    }
+
+    @Override
+    protected RestoreJobProgressFetchPolicy extractParamsOrThrow(RoutingContext context)
+    {
+        List<String> fetchPolicyValues = context.queryParam(ApiEndpointsV1.FETCH_POLICY);
+        Preconditions.checkArgument(fetchPolicyValues.size() == 1, "Must define exactly one RestoreJobProgressFetchPolicy");
+        return RestoreJobProgressFetchPolicy.fromString(fetchPolicyValues.get(0));
```

Comment:
I think it makes sense to have a default policy defined when the user does not provide one. After all, query params are optional